addons/l10n_cz/models/account_move_line.py

```diff
@@ -0,0 +1,22 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import models, fields, api
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @api.depends('currency_id', 'company_id', 'move_id.taxable_supply_date')
+    def _compute_currency_rate(self):
+        if self.company_id.country_code == "CZ":
+            for line in self:
+                if line.currency_id:
+                    line.currency_rate = self.env['res.currency']._get_conversion_rate(
+                        from_currency=line.company_currency_id,
+                        to_currency=line.currency_id,
+                        company=line.company_id,
+                        date=line.move_id.taxable_supply_date or line.move_id.invoice_date or fields.Date.context_today(line),
+                    )
+                else:
+                    line.currency_rate = 1
+        else:
+            super()._compute_currency_rate()
```

Comment:
This change is not mentioned in the commit / pr description / task
please can you add a brief description of the change to currency rate computation in those places 🙂

Since you've already gone to the effort of specifying the recomputation of the `move_id.date` based on the `move.taxable_supply_date`, there's no need to add the `taxable_supply_date` as a dependency, since a transitive dependency exists already on `move_id.date`. (see the dependencies of the original `_compute_currency_rate` method)

The other two dependencies are already declared on the original `_compute_currency_rate` method definition, so there's no need to declare them again. This means we can remove the `@api.depends`

the move's `country_code` is the fiscal country, so I've changed the condition a little bit

```suggestion
    def _compute_currency_rate(self):
        l10n_cz_lines = self.filtered(
            lambda line: (
                line.move_id.country_code == 'CZ'
                and line.move_id.date
                and line.currency_id
            )
        )
        for line in l10n_cz_lines:
            line.currency_rate = self.env['res.currency']._get_conversion_rate(
                from_currency=line.company_currency_id,
                to_currency=line.currency_id,
                company=line.company_id,
                date=line.move_id.date or fields.Date.context_today(line),
            )
        super(AccountMoveLine, (self - l10n_cz_lines))._compute_currency_rate()
```

I specified `date=line.move_id.date or fields.Date.context_today(line),` after looking at the use case and discussing with @Olivier-olma. This use case specifically implies that we would use the current date before we use the `invoice_date` (since the current day is more often the date of the taxable supply than the 'invoice date').

Reply:
Sorry, I missed mentioning that in the description.

What If the user changed the accounting date after setting the taxable_supply_date
in this case, I understood that we need to rely on taxable_supply_date not accounting date.
So, Changing the accounting date after setting the taxable_supply_date won't achieve that

I think that we need taxable_supply_date in `depends` and `_get_conversion_rate`

Reply:
Ah I see what you're getting at, in that case, we should do:
```python
date=line.move_id.taxable_supply_date or line.move_id.date or fields.Date.context_today(line),
```

however I don't think the added dependency is needed. Changing the date will trigger the `_compute_currency_rate` regardless, and changing the `taxable_supply_date` will trigger recomputation of the `date` thus triggering `_compute_currency_rate`